### PR TITLE
Python flask login

### PR DESCRIPTION
**Pull request by mooreds**
This is a blog post completing the trilogy of create a registration form, theme it, and then let folks manage the data they put in it.

### CODE REVIEW

File: site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+The code which builds a form for updating is also worth checking out:
+
+```html
+<!-- ... -->
+<form action="/update" method="post">
+  {% if registration_data %}  
+    {% for key in registration_data.keys() %}
+      <p>
+      {% if key == 'geographicarea' %}
+        Geographic area: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['geographicarea']}}' />
```
Comment (voidmain) [inline site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md:531]: Same error rendering happening here as well. See if you can fix it and if not, it should be fine.

File: site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+<!-- ... -->
+```
+
+This builds a form from the same `registration_data` array. But it does something else which you might find intriguing. Let's take a deeper look at the maximum price form field, which has some additional newlines added below:
+
+```html
+<!-- ... -->
+Maximum home price: 
+  <input 
+  type='{{fields['registration.data.'+key].control}}'
```
Comment (voidmain) [inline site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md:555]: And a few more render errors here as well.

File: site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
+  fields=[]
+  fusionauth_api_client = FusionAuthClient(app.config['API_KEY'], app.config['FA_URL'])
+  if session.get('user') != None:
+    user = session['user']
+    user_id = user['sub']
+    application_id = user['applicationId']
+
+    client_response = fusionauth_api_client.retrieve_registration(user_id, application_id)
+    if client_response.was_successful():
+      #print(client_response.success_response)
```
Comment (voidmain) [inline site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md:614]: Do we want to leave these print lines in here commented out? There are a couple in this code snippet.

File: site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md [hunk 4]
```diff
@@ -0,0 +1,815 @@
...
+  </div>
+  <div>
+  <h3>Update your preferences</h3>
+
+    <form action="/update" method="post">
+      {% if registration_data %}  
+        {% for key in registration_data.keys() %}
+          <p>
+          {% if key == 'geographicarea' %}
+            Geographic area: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['geographicarea']}}' />
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md:467]: Changed to use the jinja renderer, which helps a bit.

File: site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md (hunk 3 above)
Comment (mooreds) [reply site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md:614]: Will remove.

**Pull request merged**